docs/building/internals.md

```diff
@@ -115,22 +115,39 @@ and the NuGet package for `libpsl-native` is `libpsl`.
 
 ### psrp.windows
 
-To build `pwrshplugin.dll`, you need to install Visual Studio 2015 and run `Start-PSBootstrap -BuildWindowsNative` to install the prerequisites.
+To build `pwrshplugin.dll`, you need to install Visual Studio 2017 and run `Start-PSBootstrap -BuildWindowsNative` to install the prerequisites.
+
+Ensure the following individual components are selected:
+
+- [ ] VC++ 2017 v141 toolset (x86, x64)
+- [ ] Visual C++ compilers and libraries for ARM
+- [ ] Visual C++ compilers and libraries for ARM64
+- [ ] Visual C++ tools for CMake
+- [ ] Visual C++ ATL Support
+- [ ] Windows 10 SDK (10.0.16299.0) for Desktop C++ (ARM and ARM64)
+- [ ] Windows 10 SDK (10.0.16299.0) for Desktop C++ (x86 and x64)
+
+Ensure [CMake](https://cmake.org/download/) 3.10.0 or newer is installed which supports VS2017 and ARM64 generator.
+
 Then run `Start-BuildNativeWindowsBinaries` to build the binary.
-For example, the following builds the release flavor of the binary targeting x64 architecture.
+For example, the following builds the release flavor of the binary targeting arm64 architecture.
 
 ```powershell
-Start-BuildNativeWindowsBinaries -Configuration Release -Arch x64
+Start-BuildNativeWindowsBinaries -Configuration Release -Arch x64_arm64
 ```
 
-After that, the binary `pwrshplugin.dll` and its PDB file will be placed under 'src/powershell-win-core'.
+Be sure to build and test for all supported architectures: x86, x64, x64_arm, and x64_arm64.
+
+The `x64_arm` and `x64_arm64` architectures mean that the host system needs to be x64 to cross-compile to ARM.
+When building for multiple architectures, be sure to use the `-clean` switch as cmake will cache the previous run and the wrong compiler will be used to generate the subsequent architecutes.
+
+After that, the binary `pwrshplugin.dll` and its PDB file will be placed under 'src\powershell-native\bin\release\coreclr'.
```

Comment:
Maybe we should use the path 'src\powershell-win-core' here. The dll and pdb files will be bin-placed to "src\powershell-win-core" (see the [script](https://github.com/PowerShell/PowerShell/blob/master/build.psm1#L277)), and it's the same to 'PowerShell.Core.Instrumentation.dll' which will also be produced when running `Start-BuildNativeWindowsBinaries`.

Reply:
Ok, will change it.